src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/CodeDirectiveOnTypeFormattingTest.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting
+{
+    public class CodeDirectiveOnTypeFormattingTest : FormattingTestBase
+    {
+        [Fact]
+        public async Task CloseCurly_Class_SingleLine()
+        {
+            await RunOnTypeFormattingTestAsync(
+input: @"
+@code {
+ public class Foo{|}|
+}
+",
+expected: @"
+@code {
+    public class Foo { }
+}
+",
+triggerCharacter: "}");
+        }
+
+        [Fact]
+        public async Task CloseCurly_Class_MultiLine()
+        {
+            await RunOnTypeFormattingTestAsync(
+input: @"
+@code {
+ public class Foo{
+|}|
+}
+",
+expected: @"
+@code {
+    public class Foo
+    {
+    }
+}
+",
+triggerCharacter: "}");
+        }
+
+        [Fact]
+        public async Task CloseCurly_Method_SingleLine()
+        {
+            await RunOnTypeFormattingTestAsync(
+input: @"
+@code {
+ public void Foo{|}|
+}
+",
+expected: @"
+@code {
+    public void Foo { }
+}
+",
+triggerCharacter: "}");
+        }
+
+        [Fact]
+        public async Task CloseCurly_Method_MultiLine()
+        {
+            await RunOnTypeFormattingTestAsync(
+input: @"
+@code {
+ public void Foo{
+|}|
+}
+",
+expected: @"
+@code {
+    public void Foo
+    {
+    }
+}
+",
+triggerCharacter: "}");
+        }
+
+        [Fact]
+        public async Task CloseCurly_Property_SingleLine()
+        {
+            await RunOnTypeFormattingTestAsync(
+input: @"
+@code {
+ public string Foo{ get;set;|}|
+}
+",
+expected: @"
+@code {
+    public string Foo { get; set; }
+}
+",
+triggerCharacter: "}");
+        }
+
+        [Fact(Skip = "TBD")]
```

Comment:
Skipped a few tests here because they can't be tested without a real C# formatter. Right now, I am faking a C# formatter based on the expected text but that doesn't work well for some scenarios. Skipping these until we can use a real C# formatter for testing.
Unfortunately, Roslyn doesn't expose APIs for doing onTypeFormatting. I tried a bunch of things to maybe directly invoke their LSP handler but it was more complicated than I thought. I'll file a separate issue for building that infrastructure and link that issue here.

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/24751